libs/algorithms/include/hpx/parallel/algorithms/reduce.hpp

```diff
@@ -335,22 +202,296 @@ namespace hpx { namespace parallel { inline namespace v1 {
+    }    // namespace detail
+
     // clang-format off
     template <typename ExPolicy, typename FwdIterB, typename FwdIterE,
+        typename T, typename F,
         HPX_CONCEPT_REQUIRES_(
             execution::is_execution_policy<ExPolicy>::value &&
             hpx::traits::is_sentinel_for<FwdIterE, FwdIterB>::value
         )>
     // clang-format on
-    typename util::detail::algorithm_result<ExPolicy,
-        typename std::iterator_traits<FwdIterB>::value_type>::type
-    reduce(ExPolicy&& policy, FwdIterB first, FwdIterE last)
+    HPX_DEPRECATED(
+        "hpx::parallel::reduce is deprecated, use hpx::ranges::reduce instead")
+        typename util::detail::algorithm_result<ExPolicy, T>::type
+        reduce(ExPolicy&& policy, FwdIterB first, FwdIterE last, T init, F&& f)
+    {
+        typedef hpx::traits::is_segmented_iterator<FwdIterB> is_segmented;
+
+        return detail::reduce_(std::forward<ExPolicy>(policy), first, last,
+            std::move(init), std::forward<F>(f), is_segmented{});
+    }
+
+    // clang-format off
+    template <typename ExPolicy, typename FwdIterB, typename FwdIterE,
+        typename T,
+        HPX_CONCEPT_REQUIRES_(
+            execution::is_execution_policy<ExPolicy>::value &&
+            hpx::traits::is_sentinel_for<FwdIterE, FwdIterB>::value
+        )>
+    // clang-format on
+    HPX_DEPRECATED(
+        "hpx::parallel::reduce is deprecated, use hpx::ranges::reduce instead")
+        typename util::detail::algorithm_result<ExPolicy, T>::type
+        reduce(ExPolicy&& policy, FwdIterB first, FwdIterE last, T init)
+    {
+        typedef hpx::traits::is_segmented_iterator<FwdIterB> is_segmented;
+
+        return detail::reduce_(std::forward<ExPolicy>(policy), first, last,
+            std::move(init), std::plus<T>(), is_segmented{});
+    }
+
+    // clang-format off
+    template <typename ExPolicy, typename FwdIterB, typename FwdIterE,
+        HPX_CONCEPT_REQUIRES_(
+            execution::is_execution_policy<ExPolicy>::value &&
+            hpx::traits::is_sentinel_for<FwdIterE, FwdIterB>::value
+        )>
+    // clang-format on
+    HPX_DEPRECATED(
+        "hpx::parallel::reduce is deprecated, use hpx::ranges::reduce instead")
+        typename util::detail::algorithm_result<ExPolicy,
+            typename std::iterator_traits<FwdIterB>::value_type>::type
+        reduce(ExPolicy&& policy, FwdIterB first, FwdIterE last)
     {
         typedef typename std::iterator_traits<FwdIterB>::value_type value_type;
 
         typedef hpx::traits::is_segmented_iterator<FwdIterB> is_segmented;
 
         return detail::reduce_(std::forward<ExPolicy>(policy), first, last,
-            value_type(), std::plus<value_type>(), is_segmented());
+            value_type{}, std::plus<value_type>(), is_segmented{});
     }
 }}}    // namespace hpx::parallel::v1
+
+namespace hpx {
+
+    ///////////////////////////////////////////////////////////////////////////
+    // CPO for hpx::copy_n
```

Comment:
```suggestion
    // CPO for hpx::reduce
```

Reply:
@hkaiser I wanted to try this PR out and fixed this and a couple headers while doing so.